activity_calendar/templates/activity_calendar/activity_page_no_slots.html

```diff
@@ -0,0 +1,79 @@
+{% extends "activity_calendar/activity_page_base.html" %}
+{% load calendar_tags %}
+{% load to_extended_user %}
+
+{% block local-messages %}
+    {# empty, this will be locally handled through Django messages framework #}
+{% endblock local-messages %}
+
+{% block total_participants %}
+    {# We don't use that on this layout #}
+{% endblock total_participants %}
+
+{% block sign-up-status %}
+    {% if is_subscribed %}
+        <div class="card border border-success greenify activity-wide">
+            <div class="row">
+                <div class="col-12 col-sm-7 col-md-12">
+                    <div class="p-1">
+                        You are registered for this activity
+                    </div>
+                </div>
+                <div class="col-12 col-sm-5 col-md-12">
+                    {% if form.is_valid %}
+                        <form method="post">
+                            {% csrf_token %}
+                            {{ form }}
+                            <input type="submit" class="btn btn-danger btn-subscribe btn-block" value="Deregister">
+                        </form>
+                    {% else %}
+                        <button type="button" class="btn btn-outline-danger btn-block disabled" data-container="body" data-toggle="popover" data-placement="bottom"
+                                data-content="{% form_rejection_reason form %}">
+                            Deregister
+                        </button>
+                    {% endif %}
+                </div>
+            </div>
+        </div>
+    {% else %}
+        <div class="card border border-danger activity-wide">
+            <div class="row">
+                <div class="col-12 col-sm-7 col-md-12">
+                    <div class="d-block p-2">
+                        You are not yet registered for this activity
+                    </div>
+                </div>
+                <div class="col-12 col-sm-5 col-md-12">
+                    {% if form.is_valid %}
+                        <form method="post">
+                            {% csrf_token %}
+                            {{ form }}
+                            <input type="submit" class="btn btn-success btn-block" value="Register">
+                        </form>
+                    {% else %}
+                        <button type="button" class="btn btn-outline-success btn-block disabled" data-container="body" data-toggle="popover" data-placement="bottom"
+                                data-content="{% form_rejection_reason form %}">
+                            Register
+                        </button>
+                    {% endif %}
+                </div>
+            </div>
+        </div>
+    {% endif %}
+{% endblock sign-up-status %}
+
+{% block additional-contents %}
+    {% if request.user.is_authenticated %}
```

Comment:
Currently, only members* can view who registered for a slot. This implicitly reverts that change.

*Members here means "users who are in Squire's membership file". Anyone can create a Squire account, and there's no need for (possibly) random people to see who's registered.

Reply:
What's the way we can check on a user if that user is member?
And more importantly, does the request given member default to ExtendedUser or just User? Otherwise we might need to patch that in order to prevent code injection all over the place to interpret Users as ExtendedUsers (ugly), I made an issue about that a while ago I believe.

Reply:
There's the `to_member` template filter which has been used in two other places, which converts a user (or extendeduser) to a MemberUser. A MemberUser is also an ExtendedUser so there's no need for _another_ (ugly) conversion.

There's indeed an open issue about it already (#66) and it is something we should fix.